osu.Game/Online/Chat/ChatLink.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Cursor;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Overlays;
+using System;
+using System.Text.RegularExpressions;
+
+namespace osu.Game.Online.Chat
+{
+    public class ChatLink : OsuSpriteLink, IHasTooltip
+    {
+        private APIAccess api;
+        private BeatmapSetOverlay beatmapSetOverlay;
+        private ChatOverlay chat;
+
+        public override bool HandleInput => !string.IsNullOrEmpty(Url);
+
+        // 'protocol' -> 'https', 'http', 'osu', 'osump' etc.
+        // 'content' -> everything after '<protocol>://'
+        private Match getUrlMatch() => Regex.Match(Url, @"^(?<protocol>osu(?:mp)?|https?):\/\/(?<content>.*)");
+
+        protected override void OnLinkClicked()
+        {
+            var urlMatch = getUrlMatch();
+            if (urlMatch.Success)
+            {
+                var args = urlMatch.Groups["content"].Value.Split('/');
+
+                switch (urlMatch.Groups["protocol"].Value)
+                {
+                    case "osu":
+                        if (args.Length == 1)
+                        {
+                            base.OnLinkClicked();
+                            break;
+                        }
+
+                        switch (args[0])
+                        {
+                            case "chan":
+                                var foundChannel = chat.AvailableChannels.Find(channel => channel.Name == args[1]);
+
+                                // links should be filtered out by now if a channel doesn't exist
+                                chat.OpenChannel(foundChannel ?? throw new ArgumentException($"Unknown channel name ({args[1]})."));
+                                break;
+                            case "edit":
+                                chat.Game?.LoadEditorTimestamp();

```

Comment:
You should not be accessing game through chat. Please DI game if you need game.

Also you'll want to make the BDL method nullable as game won't always be available.